Fix environment variables to correctly data format

naonishijima opened the pull request:
process.env will implicitly convert the value to a string.
If an environment value is set as a boolean or number, the value
will be a string. This patch uses environment value from nconf.env
not process.env.

Signed-off-by: Nao Nishijima <nao.nishijima.xt@hitachi.com>
Reported-by: Shinsuke Hasegawa <shinsuke.hasegawa.fc@hitachi.com>

Resolves #553

bestbeforetoday commented:
It looks like the build for this PR git hit by a bug in typescript-eslint, which I have worked around in PR #555. If you rebase on the latest **main** branch, the build should be happier.

bestbeforetoday commented:
I wonder if the mapping of environment variables could be simplified by just passing a transform function to the `nconf.env()` call, as described in the [nconf documentation](https://github.com/indexzero/nconf#env). What do you think?

A unit test that would fail with the existing implementation but passes with the changes introduced by this PR would be good, to confirm the change solves the problem and ensure that nobody accidentally introduces a regression in the future.

naonishijima committed:
Fix environment variables to correctly data format

process.env will implicitly convert the value to a string.
If an environment value is set as a boolean or number, the value
will be a string. This patch uses environment value from nconf.env
not process.env.

Signed-off-by: Nao Nishijima <nao.nishijima.xt@hitachi.com>
Reported-by: Shinsuke Hasegawa <shinsuke.hasegawa.fc@hitachi.com>
